ci-operator/step-registry/upi/install/azurestack/upi-install-azurestack-commands.sh

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+export HOME=/tmp
+
+if [[ -z "$RELEASE_IMAGE_LATEST" ]]; then
+  echo "RELEASE_IMAGE_LATEST is an empty string, exiting"
+  exit 1
+fi
+
+# Ensure ignition assets are configured with the correct invoker to track CI jobs.
+export OPENSHIFT_INSTALL_INVOKER="openshift-internal-ci/${JOB_NAME_SAFE}/${BUILD_ID}"
+export TEST_PROVIDER='azurestack'
+export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=quay.io/padillon/ash-release-mirror:latest
+echo $OPENSHIFT_INSTALL_INVOKER
+
+dir=/tmp/installer
+mkdir "${dir}"
+pushd "${dir}"
+cp -t "${dir}" \
+    "${SHARED_DIR}/install-config.yaml"
+
+if ! pip -V; then
+    echo "pip is not installed: installing"
+    if python -c "import sys; assert(sys.version_info >= (3,0))"; then
+      python -m ensurepip --user || easy_install --user 'pip'
+    else
+      echo "python < 3, installing pip<21"
+      python -m ensurepip --user || easy_install --user 'pip<21'
+    fi
+fi
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o ${HOME}/yq && chmod +x ${HOME}/yq
+PATH="${HOME}:${HOME}/.local/bin:${PATH}"
+
+cat "${SHARED_DIR}/servicePrincipal.json" | jq '{subscriptionId:.subscriptionId,clientId:.appId,clientSecret:.password,tenantId:.tenant}' \
+     > "${SHARED_DIR}/osServicePrincipal.json"
+
+CLUSTER_NAME=$(yq r install-config.yaml metadata.name)
+AAD_CLIENT_ID=$(jq -r .clientId ${SHARED_DIR}/osServicePrincipal.json)
+AZURE_REGION=ppe3
+SSH_KEY=$(yq r install-config.yaml sshKey | xargs)
+BASE_DOMAIN=$(yq r install-config.yaml baseDomain)
+TENANT_ID=$(jq -r .tenantId ${SHARED_DIR}/osServicePrincipal.json)
+AAD_CLIENT_SECRET=$(jq -r .clientSecret ${SHARED_DIR}/osServicePrincipal.json)
+APP_ID=$(jq -r .clientId ${SHARED_DIR}/osServicePrincipal.json)
+
+export CLUSTER_NAME
+export AAD_CLIENT_ID
+export AZURE_REGION
+export SSH_KEY
+export TENANT_ID
+export BASE_DOMAIN
+export AAD_CLIENT_SECRET
+export APP_ID
+
+az cloud register \
+    -n PPE \
+    --endpoint-resource-manager "https://management.ppe3.stackpoc.com" \
+    --suffix-storage-endpoint "ppe3.stackpoc.com" 
+az cloud set -n PPE
+az cloud update --profile 2019-03-01-hybrid
+az login --service-principal -u $APP_ID -p $AAD_CLIENT_SECRET --tenant $TENANT_ID > /dev/null
+
+# shellcheck disable=SC2034
+SUBSCRIPTION_ID=$(az account show | jq -r .id)
+export SUBSCRIPTION_ID
+export AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"
+
+echo "Installing python modules: yaml"
+python3 -c "import yaml" || pip3 install --user pyyaml
+
+# remove workers from the install config so the mco won't try to create them
+python3 -c '
+import yaml;
+path = "install-config.yaml";
+data = yaml.full_load(open(path));
+data["compute"][0]["replicas"] = 0;
+open(path, "w").write(yaml.dump(data, default_flow_style=False))'
```

Comment:
I made an earlier comment about setting this in the install-config. We can hold off on a change until we standardize what we are doing in the upi docs and then mirror it here. So it is fine for the time being.